### ISSUE TITLE
With revealjs, plotly tooltip not aligned correctly

### ISSUE THREAD

**Issue by lizwillow** [quarto-dev/quarto-cli#2430]
### plotly tooltip misaligned in quarto with revealjs

Rendering the following example results in an html file where the tooltip is off-center. You would need to position the pointer above and to the left of the point you want to select in order to view the information for that data point. I tried opening it in Microsoft Edge and Google Chrome and both had this issue.

````md
---
title: "example"
author: "tmp"
format:
  revealjs
---

## slide 1

```{r}
library(plotly)
plot_ly(data = iris, x = ~Sepal.Length, y = ~Petal.Length)
```
````

Additional information:
Version of quarto: 1.1.189
Operating system: Windows 10 Enterprise
IDE you’re using, and its version: RStudio version 2022.7.1.554

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
I'm pretty sure this is happening because the tooltip is using units that are getting affected by the `transform` node that the revealjs format uses. Unfortunately, it means that it will be hard for us to solve this in general.

@jjallaire This is a big change, but I think we should consider forking revealjs and reworking their setup so we never have a `transform` node in the DOM. It wreaks havoc with a ton of downstream library expectations.

**Comment by jjallaire** [comment]
The counterpoint to this is that the `transform` is kind of Reveal's secret sauce because it allows you to target a known set of pixel dimensions with your content (no matter how the viewport is resized). It's also extremely important to be able to target images, arrows, illustrations, etc. with absolute positions. Interestingly for a lot of people used to having this level of control in PowerPoint, etc. Reveal is the first HTML slides framework that gives them this control and they love it. I could be missing something here but given all this I'm not sure we'll ever get away from the `transform` w/o losing some of the rest of Reveal's benefits.

**Comment by cscheid** [comment]
I see, that makes sense, thanks.

The issue comes from HTML libraries that love to inject transient DOM nodes (tooltips, for example; I see the same thing with mermaidjs) at the end of the body or somewhere else where they try to get absolute positioning. Then, the fact that they inject the DOM node in a place with a different transform makes it impossible to line them up.

I wonder if, instead, we can make the transform node take _more_ of the DOM, so that these libraries can't accidentally escape it.

**Comment by cderv** [comment]
Just to share some references as we had related issue in the R package at some point and it was fixed in plotly.js directly. 
See more details in https://github.com/rstudio/revealjs/issues/47#issuecomment-921770171 and the plotly.js PR : https://github.com/plotly/plotly.js/pull/5193

New plotly.js has been available in plotly R package v4.10.0

However, it seems a new issue already reported in plotly and reveal.js 

* https://github.com/plotly/plotly.js/issues/6169
* https://github.com/hakimel/reveal.js/issues/3208

In short, `zoom` attributes is used for the slide scaling system, and it creates the tooltip issue. 

I believe this has been fixed upstream in reveal.js already and `zoom` is no more used. See https://github.com/hakimel/reveal.js/commit/e281b3234e7991283ce4dcca705dd9a6a9ebe5d2

Possibly using this commit or master for our embed reveal.js, or patching it will solve the issue here..

**Comment by cderv** [comment]
I confirm that using the https://github.com/hakimel/reveal.js/commit/e281b3234e7991283ce4dcca705dd9a6a9ebe5d2 for the revealjs version, change the scaling and fix the issue. 

In HTML we had 
````html
<div class="slides" style="width: 1050px; height: 700px; zoom: 1.69843;">
<!-- slides in there-->
</div>
````
which becomes
````html
<div class="slides" style="width: 1050px; height: 700px; inset: 50% auto auto 50%; transform: translate(-50%, -50%) scale(1.69843);">
<!-- slides in there-->
</div>
````

See the PR #2437 for a branch with the updated revealjs that fixed the issue. I share this as an example that upstream seems to have fixed this scaling issue.

**Comment by timbmg** [comment]
I still seem to experience this issue with vega-altair plots

**Comment by cderv** [comment]
@timbmg this is highly possible as the original issue was a conflict in revealjs and plotly. Both being JS libraries with there own feature and configuration. 

It is possible that this is the same for vega-altair too. 

Can you open a new issue to track ? 

Investigation will be the following then: 
* Is this an issue in RevealJS directly for supporting some feature of Vega Altair ? 
* If not, what does Quarto do that create the issue ? 

Thanks.
